tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -727,8 +727,16 @@ static void test_valid_function_without_alarm(const struct device *dev)
 		ticks_expected = counter_us_to_ticks(dev, wait_for_us);
 	}
 
+#ifdef CONFIG_COUNTER_MCUX_LPC_RTC_HIGHRES
```

Comment:
I don't think we should have device-specific `#ifdef` statements burrowed this deep in test cases. Better to generalize the condition.

Reply:
Any recommendations on how we should generalize?

Reply:
I'd suggest using a generic "inaccuracy" #define at the top of the file similar as to how the other customisations of this test suite is done.